lib/dao.js

```diff
@@ -2632,6 +2632,8 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
           'replaceById() must be implemented by the connector');
 
   var pkName = idName(this);
+  var initialData = {};
+  copyData(data, initialData);
```

Comment:
Hmm, creating a copy of all data is rather expensive operation. Could you please take another look and check if there is a way how to avoid that?

For example, take a look a the following line - is it necessary to always fill in primary key?

```js
if (!data[pkName]) data[pkName] = id;
```